### PR TITLE
docs: fix install commands for linux distribution

### PR DESCRIPTION
**Pull request by akshaysngupta**
Commit Message: Fix install commands for linux distribution
Additional Description:
Risk Level: Low
Testing: manual testing
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: docs/root/start/install.rst [hunk 1]
```diff
@@ -19,16 +19,16 @@ Install Envoy on Debian-based Linux
 
    .. code-tab:: console Debian bookworm
 
-      $ wget https://apt.envoyproxy.io/signing.key  | sudo gpg --dearmor -o /etc/apt/keyrings/envoy-keyring.gpg
```
Comment (phlax) [inline docs/root/start/install.rst:22]: `wget` is more commonly installed in debian-based distros

File: docs/root/start/install.rst [hunk 2]
```diff
@@ -19,16 +19,16 @@ Install Envoy on Debian-based Linux
...
       $ sudo apt-get update
       $ sudo apt-get install envoy
       $ envoy --version
 
    .. code-tab:: console Debian bullseye
 
-      $ wget https://apt.envoyproxy.io/signing.key  | sudo gpg --dearmor -o /etc/apt/keyrings/envoy-keyring.gpg
-      $ echo "deb[signed-by=/etc/apt/keyrings/envoy-keyring.gpg] https://apt.envoyproxy.io bullseye main" | sudo tee /etc/apt/sources.list.d/envoy.list
+      $ curl -fsSL https://apt.envoyproxy.io/signing.key | sudo gpg --dearmor -o /etc/apt/keyrings/envoy-keyring.gpg
+      $ echo "deb [signed-by=/etc/apt/keyrings/envoy-keyring.gpg] https://apt.envoyproxy.io bullseye main" | sudo tee /etc/apt/sources.list.d/envoy.list
```
Comment (phlax) [inline docs/root/start/install.rst:31]: is this change necessary?

**Review by phlax — COMMENTED** [review]
im not sure this fixes anything - wget is more common to a base system - altho tbf can be absent from a totally barebones system

curl pretty much **always** must be installed

how about instead adding install lines for 

- wget
- sudo
- gpg

as these are the requirements

File: docs/root/start/install.rst (hunk 1 above)
Comment (akshaysngupta) [reply docs/root/start/install.rst:22]: `wget` doesn't pass the content to `stdout` as needed by `gpg`. The file gets stored locally and doesn't get processed by `gpg`. When running `apt update`, it wouldn't find the public key and fail.


Following can work as well:
```bash
wget -O /tmp/signing.key https://apt.envoyproxy.io/signing.key
sudo gpg --dearmor -o /etc/apt/keyrings/envoy-keyring.gpg /tmp/signing.key
```

File: docs/root/start/install.rst (hunk 2 above)
Comment (akshaysngupta) [reply docs/root/start/install.rst:31]: installation was failing here as well.
```
E: Type 'deb[signed-by=/etc/apt/keyrings/envoy-keyring.gpg]' is not known on line 1 in source list /etc/apt/sources.list.d/envoy.list
E: The list of sources could not be read.
```

That space is required to fix this.

File: docs/root/start/install.rst (hunk 1 above)
Comment (phlax) [reply docs/root/start/install.rst:22]: argh - i thought i tested this 8/

Comment (phlax) [reply docs/root/start/install.rst:22]: i think lets do

```console
wget -O- https://apt.envoyproxy.io/signing.key | sudo gpg --dearmor -o /etc/apt/keyrings/envoy-keyring.gpg

```

File: docs/root/start/install.rst (hunk 2 above)
Comment (phlax) [reply docs/root/start/install.rst:33]: apologies - my bad - you are right

**Review by phlax — APPROVED** [review]
lgtm, thanks @akshaysngupta - appreciated
